src/services/permissions-checker.js

```diff
@@ -29,7 +32,81 @@ function PermissionsChecker(
     return allowed && (!users || users.includes(parseInt(userId, 10)));
   }
 
-  function isAllowed() {
+  // NOTICE: Compute a scope to replace $currentUser variables with
+  //         the actual user values. This will generate the expected
+  //         conditions filters when applied on the server scope response
+  function computeConditionFiltersFromScope(userId, collectionListScope) {
+    const computedConditionFilters = _.clone(collectionListScope.filter);
+    computedConditionFilters.conditions.forEach((condition) => {
+      if (condition.value
+        && condition.value.startsWith('$')
+        && collectionListScope.dynamicScopesValues.users[userId]) {
+        condition.value = collectionListScope
+          .dynamicScopesValues
+          .users[userId][condition.value];
+      }
+    });
+    return computedConditionFilters;
+  }
+
+  // NOTICE: Check if `expectedConditionFilters` at least contains a definition of
+  //         `actualConditionFilters`
+  function isConditionFromScope(expectedFilterConditions, actualFilterCondition) {
+    return expectedFilterConditions.filter((expectedCondition) =>
+      expectedCondition.value === actualFilterCondition.value
+      && expectedCondition.operator === actualFilterCondition.operator
+      && expectedCondition.field === actualFilterCondition.field).length > 0;
+  }
+
+  async function isCollectionListAllowed(collectionListScope) {
+    if (!collectionListScope) {
+      return true;
+    }
+
+    try {
+      const expectedConditionFilters = computeConditionFiltersFromScope(
+        collectionListRequest.userId,
+        collectionListScope,
+      );
+
+      const formatAggregation = (aggregator, conditions) => {
+        const filtredConditions = conditions.filter(Boolean);
+        // NOTICE: Exit case - filtredConditions[0] should be the scope
+        if (filtredConditions.length === 1
+          && filtredConditions[0].aggregator
+          && aggregator === 'and') {
+          return filtredConditions[0];
+        }
+
+        // NOTICE: During the tree travel, check if `conditions` & `aggregator`
+        //         match with expectations
+        return filtredConditions.length === expectedConditionFilters.conditions.length
+          && (aggregator === expectedConditionFilters.aggregator || aggregator === 'and')
+          ? { aggregator, conditions: filtredConditions }
+          : null;
+      };
+
+      const formatCondition = (condition) =>
+        isConditionFromScope(expectedConditionFilters.conditions, condition);
+      // NOTICE: Perform a travel to find the scope in filters
+      const scopeFound = await perform(
+        collectionListRequest.filters,
+        formatAggregation,
+        formatCondition,
+      );
+      if (expectedConditionFilters.conditions.length === 1) {
```

Comment:
I know why you do that, but you can add a notice for the next team mate who gonna work on it 😉